TradeItIosEmsApi/TradeItConnector.m

```diff
@@ -413,48 +400,42 @@ -(void) sendEMSRequest:(NSMutableURLRequest *)request
+        NSURLSession *session = [NSURLSession sharedSession];
+        [[session dataTaskWithRequest:request completionHandler:^(NSData * _Nullable data, NSURLResponse * _Nullable response, NSError * _Nullable error) {
+            NSHTTPURLResponse *httpResponse = (NSHTTPURLResponse *) response;
+            if ((data == nil) || ([httpResponse statusCode] != 200)) {
+                //error occured
+                NSLog(@"ERROR from EMS server response=%@ error=%@", response, error);
+                TradeItErrorResult *errorResult = [TradeItErrorResult tradeErrorWithSystemMessage:@"error sending request to ems server"];
+                dispatch_async(dispatch_get_main_queue(), ^(void){
+                    completionBlock(errorResult, nil);
+                });
+
+                return;
+            }
 
-        /*
-        NSLog(@"----------Response %@----------", [[request URL] absoluteString]);
-        NSLog(jsonResponse);
-        */
+            NSMutableString *jsonResponse = [[NSMutableString alloc] initWithData:data encoding:NSUTF8StringEncoding];
 
-        //first convert to a generic result to check the type
-        TradeItResult *tradeItResult = [TradeItJsonConverter buildResult:[TradeItResult alloc]
-                                                               jsonString:jsonResponse];
-        
-        if ([tradeItResult.status isEqual:@"ERROR"]) {
-            TradeItErrorResult * errorResult;
-            
-            if (![tradeItResult isKindOfClass:[TradeItErrorResult class]]) {
-                errorResult = (TradeItErrorResult *)[TradeItJsonConverter buildResult:[TradeItErrorResult alloc]
-                                                                           jsonString:jsonResponse];
-            } else {
-                errorResult = (TradeItErrorResult *) tradeItResult; //this type of error caused by something wrong parsing the response
+//             NSLog(@"----------Response %@----------", [[request URL] absoluteString]);
+//             NSLog(jsonResponse);
+
+            //first convert to a generic result to check the type
+            TradeItResult *tradeItResult = [TradeItJsonConverter buildResult:[TradeItResult alloc]
+                                                                  jsonString:jsonResponse];
+
+            if ([tradeItResult.status isEqual:@"ERROR"]) {
+                TradeItErrorResult * errorResult;
+
+                if (![tradeItResult isKindOfClass:[TradeItErrorResult class]]) {
+                    errorResult = (TradeItErrorResult *)[TradeItJsonConverter buildResult:[TradeItErrorResult alloc]
+                                                                               jsonString:jsonResponse];
+                } else {
+                    errorResult = (TradeItErrorResult *) tradeItResult; //this type of error caused by something wrong parsing the response
+                }
+                tradeItResult = errorResult;
```

Comment:
This error code is very confusing. Where do we convert other `TradeItResult` types in to their subclasses?